### PR TITLE
Initial Scaffolding for Offline Mode

### PR DESCRIPTION
**Pull request by manuelroemer**
> ⚠ **Warning:**  
> Please don't merge before [this PR](https://github.com/openmrs/openmrs-esm-core/pull/96) has been merged!
> The changes here depend on work being done in `esm-core`.

This PR scaffolds the required parts for future offline support of the patient registration MF.
At the moment, it introduces the required webpack plugins for generating the build manifest,
defines the `setupOpenMRSOffline` function and displays placeholder content when the page is
running offline.

The following images show the `openmrs/spa/patient-registration` page in online and offline mode:

online
offline

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -14,62 +15,73 @@ const cssLoader = {
...
-          system: false,
-        },
-      },
-      {
-        test: /\.m?(js|ts|tsx)$/,
-        exclude: /(node_modules|bower_components)/,
-        use: {
-          loader: 'babel-loader',
+module.exports = env => {
+  const filename = 'openmrs-esm-patient-registration-app.js';
```
Comment (FlorianRappl) [inline webpack.config.js]: There is no need to change the layout of the whole file for this variable. The variable could be also placed module-global. No need to define it inside the function.

**Review by FlorianRappl — COMMENTED** [review]
I think this is totally fine. Great job!

File: webpack.config.js (hunk 1 above)
Comment (manuelroemer) [reply webpack.config.js]: Good point, I will change that.

**Review by FlorianRappl — APPROVED** [review]
LGTM :beers:

**Review by tejasdeshpande100 — COMMENTED** [review]
👌🏻 Awesome

**Pull request merged**